tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import Link from 'amo/components/Link';
+import UserProfileEdit, {
+  UserProfileEditBase,
+} from 'amo/components/UserProfileEdit';
+import {
+  editUserAccount,
+  fetchUserAccount,
+  getCurrentUser,
+  loadUserAccount,
+} from 'amo/reducers/users';
+import { USERS_EDIT } from 'core/constants';
+import { ErrorHandler } from 'core/errorHandler';
+import ErrorList from 'ui/components/ErrorList';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import {
+  createFakeEvent,
+  createStubErrorHandler,
+  createUserAccountResponse,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function defaultUserProps(props = {}) {
+    return {
+      display_name: 'Matt MacTofu',
+      userId: 500,
+      username: 'tofumatt',
+      ...props,
+    };
+  }
+
+  function syncPropsAndParamsUsername(username) {
+    return {
+      params: { username },
+      store: dispatchSignInActions({
+        userProps: defaultUserProps({ username }),
+      }).store,
+    };
+  }
+
+  function renderUserProfileEdit({
+    i18n = fakeI18n(),
+    params = { username: 'tofumatt' },
+    store = dispatchSignInActions({
+      userProps: defaultUserProps(),
+    }).store,
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <UserProfileEdit i18n={i18n} params={params} store={store} {...props} />,
+      UserProfileEditBase,
+      { shallowOptions: { context: { i18n } } }
+    );
+  }
+
+  function mountUserProfileEdit({
+    errorHandler = createStubErrorHandler(),
+    i18n = fakeI18n(),
+    params = { username: 'tofumatt' },
+    store = dispatchSignInActions({
+      userProps: defaultUserProps(),
+    }).store,
+    ...props
+  } = {}) {
+    return mount(
+      <Provider store={store}>
+        <UserProfileEdit
+          errorHandler={errorHandler}
+          i18n={i18n}
+          params={params}
+          store={store}
+          {...props}
+        />
+      </Provider>,
+      { context: { i18n } }
+    );
+  }
+
+  it('renders user profile page', () => {
+    const root = renderUserProfileEdit();
+
+    expect(root.find('.UserProfileEdit')).toHaveLength(1);
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const username = 'i-am-not-tofumatt';
+
+    const root = renderUserProfileEdit({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount action if there is no username', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
```

Comment:
Because we don't care about the actual username for this test, just that no username is passed into render, why not make this state the default state and then we can just call `renderUserProfileEdit` below without doing this first and without passing in a `store`?